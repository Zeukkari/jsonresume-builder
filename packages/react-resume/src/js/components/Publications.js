import React from 'react'

import { Box } from 'grommet'

import {
  ResumeSection,
  SectionTitle,
  formatDate,
  ResumeDateSection,
  ResumeHeading,
  ResumeSubHeading,
  ResumeParagraph,
} from './common'

const Publications = ({ publications }) => {
  const renderItems = publications.map(item => {
    return (
      <Box key={item.company}>
        <ResumeHeading>{item.name}</ResumeHeading>
        <ResumeSubHeading>{item.publisher}</ResumeSubHeading>
        <ResumeDateSection>{formatDate(item.releaseDate)}</ResumeDateSection>
        <ResumeParagraph>{item.summary}</ResumeParagraph>
      </Box>
    )
  })
  return (
    <Box margin="none" pad="none" border="all">
      <ResumeSection>
        <SectionTitle>Publications</SectionTitle>
        <Box>{renderItems}</Box>
      </ResumeSection>
    </Box>
  )
}

Publications.defaultProps = {
  publications: [],
}

export { Publications }
