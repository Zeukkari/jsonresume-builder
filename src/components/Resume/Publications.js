import React from 'react'

import { Grommet, DataTable, Heading, Text } from 'grommet'

import { ResumeSection, SubSection } from './common'

const columns = [
  {
    property: 'name',
    header: <Text>Publication</Text>,
    primary: true,
  },
  {
    property: 'publisher',
    header: 'Company',
  },
  {
    property: 'releaseDate',
    header: 'Release Date',
  },
  {
    property: 'website',
    header: 'Website',
  },
  {
    property: 'summary',
    header: 'Summary',
  },
]

const References = ({ publications }) => {
  const renderItems = publications

  return (
    <ResumeSection>
      <SubSection>
        <Heading> Publications </Heading>
      </SubSection>
      <Grommet>
        <SubSection>
          <DataTable columns={columns} data={renderItems} />
        </SubSection>
      </Grommet>
    </ResumeSection>
  )
}

References.defaultProps = {
  volunteer: [],
}

export default References
