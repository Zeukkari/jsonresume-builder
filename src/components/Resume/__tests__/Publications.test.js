import React from 'react';
import renderer from 'react-test-renderer';

import Publications from '../Publications';

import defaultResume from '../../../util/defaultResume';


test('renders publications section', () => {
  const data = defaultResume.publications;
  const component = renderer.create(
    <Publications publications={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});