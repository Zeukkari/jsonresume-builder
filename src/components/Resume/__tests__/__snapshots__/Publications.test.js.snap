// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders publications section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 DOiol"
>
  <table
    className="StyledDataTable-xrlyjm-0 kTrXKK"
  >
    <thead
      className="StyledDataTable__StyledDataTableHeader-xrlyjm-3 bAIczD StyledTable__StyledTableHeader-sc-1m3u5g-4 dHaGyd"
    >
      <tr
        className="StyledDataTable__StyledDataTableRow-xrlyjm-1 ezpFws StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Publication
            </span>
          </div>
        </th>
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Company
            </span>
          </div>
        </th>
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Release Date
            </span>
          </div>
        </th>
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Website
            </span>
          </div>
        </th>
        <th
          className="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 folWQp"
          >
            <span
              className="StyledText-sc-1sadyjn-0 fWSbXS"
            >
              Summary
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      className="StyledDataTable__StyledDataTableBody-xrlyjm-2 hbZQKY"
    />
  </table>
</div>
`;
